File: packages/schemas/src/tag.ts
import z from "zod";

export const tagSchema = z.object({
  id: z.number().positive("id must be positive"),
  name: z
    .string()
    .trim()
    .min(3, "name must be at least 3 characters")
    .max(255, "name must be at most 255 characters"),
});

File: internal/schema/tag.go
package schema

import (
	z "github.com/Oudwins/zog"
)

type CreateTag struct {
	Name string
}

type RenameTag struct {
	Name string
}

var tagSchema = z.Struct(z.Shape{
	"name": z.String().Required(z.Message("name is required.")).
		Min(1, z.Message("name must be at least 1 character.")).
		Max(255, z.Message("name must be at most 255 characters.")),
})

var CreateTagInput = tagSchema.Pick("name")
var RenameTagInput = tagSchema.Pick("name")
